lib/server/monalisa-synchronization/MonALISAInterface.js

```diff
@@ -0,0 +1,169 @@
+ *
+ * This software is distributed under the terms of the GNU General Public
+ * License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+ *
+ * In applying this license CERN does not waive the privileges and immunities
+ * granted to it by virtue of its status as an Intergovernmental Organization
+ * or submit itself to any jurisdiction.
+ */
+
+const fs = require('fs').promises;
+const { ServicesConfig: { monalisa: monalisaConfiguration } } = require('../../config');
+const { Log } = require('@aliceo2/web-ui');
+const https = require('https');
+const { extractLhcPeriod } = require('../utilities/extractLhcPeriod');
+
+/**
+ * @typedef MonALISADataPass
+ * @param {string} name
+ * @param {number} reconstructed_events
+ * @param {string} description
+ * @param {number} output_size
+ * @param {number} last_run
+ */
+
+/**
+ * @typedef DataPassDetails
+ * @param {number[]} runNumbers
+ */
+
+/**
+ * Interface with MonALISA
+ */
+class MonALISAInterface {
+    /**
+     * Constructor
+     * @param {object} configuration configuration of synchronizer
+     * @param {URL|string} [configuration.dataPassesUrl] url to fetch data passes
+     * @param {URL|string} [configuration.dataPassDetailsUrl] url to fetch data pass details
+     * @param {string|string[]|Buffer|Buffer[]|Object[]} [configuration.pfxCertificateBytes] PFX or PKCS12 encoded private key
+     * and certificate chain @see tls.createSecureContext([options])
+     * @param {string} [configuration.certificatePassphrase] passphrase to the certificate
+     * @param {number} [configuration.yearLowerLimit] indicates how old data are accepted,
+     * year of lhc period that given data pass belongs must be greater or equal
+     */
+    constructor({
+        dataPassesUrl,
+        dataPassDetailsUrl,
+        pfxCertificateBytes,
+        certificatePassphrase,
+        yearLowerLimit,
+    } = {}) {
+        this.logger = new Log(this.constructor.name);
+        this.dataPassesUrl = new URL(dataPassesUrl);
+        this.dataPassDetailsUrl = new URL(dataPassDetailsUrl);
+        this.yearLowerLimit = yearLowerLimit;
+
+        this.httpOptions = {
+            headers: {
+                Accept:	'application/json;charset=utf-8',
+                Connection:	'keep-alive',
+            },
+            agent: pfxCertificateBytes && certificatePassphrase
+                ? new https.Agent({ pfx: pfxCertificateBytes, passphrase: certificatePassphrase }) : undefined,
+        };
+    }
+
+    /**
+     * Get data passes with valid names from MonALISA
+     * @return {DataPass[]} data passes
+     */
+    async getDataPasses() {
+        const payload = await this.fetchDataPasses();
+        return this.processDataPassesPayload(payload);
+    }
+
+    /**
+     * Fetch data passes from MonALISA
+     * @return {object<string, Partial<MonalisaDataPass>>} payload
+     */
+    async fetchDataPasses() {
+        const { default: fetch } = await import('node-fetch');
```

Comment:
I think you can directly use require for this.
I don't remember why I did use async import

Reply:
a require gives `Error [ERR_REQUIRE_ESM]: require() of ES Module /usr/src/app/node_modules/node-fetch/src/index.js not supported.`

Reply:
Ah right, so this was the reason